prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -594,31 +587,70 @@ class ReportFunction : Logging {
                 return sb.toString()
             }
 
-            fun writeConsolidatedArray(field: String, array: List<ResultDetail>) {
-                val rowsByGroupingId = hashMapOf<String, MutableList<Int>>()
-                val messageByGroupingId = hashMapOf<String, String>()
+            fun createPropertiesByGroupingId(array: List<ResultDetail>): GroupedProperties {
+                val itemsByGroupingId = mutableMapOf<String, MutableList<Int>>()
+                val messageByGroupingId = mutableMapOf<String, String>()
+                val scopesByGroupingId = mutableMapOf<String, String>()
+                val itemDetailsWithGroupingId = mutableMapOf<String, MutableList<Int>>()
                 array.forEach { resultDetail ->
                     val groupingId = resultDetail.responseMessage.groupingId()
-                    if (!rowsByGroupingId.containsKey(groupingId)) {
-                        rowsByGroupingId[groupingId] = mutableListOf()
+                    if (!itemsByGroupingId.containsKey(groupingId)) {
+                        itemsByGroupingId[groupingId] = mutableListOf()
                         messageByGroupingId[groupingId] = resultDetail.responseMessage.detailMsg()
+                        scopesByGroupingId[groupingId] = resultDetail.scope.toString()
+                        itemDetailsWithGroupingId[groupingId] = mutableListOf()
                     }
                     if (resultDetail.row != -1) {
                         // Add 2 to account for array offset and csv header
-                        rowsByGroupingId[groupingId]?.add(resultDetail.row + 1)
+                        itemsByGroupingId[groupingId]?.add(resultDetail.row + 2)
+                        itemDetailsWithGroupingId[groupingId]?.add(resultDetail.row + 2)
```

Comment:
Change to add 1 in order to normalize hl7 and csv results. We'll have to trust that it's clear enough that the header isn't included.